smolyay/grid.py

```diff
@@ -2,348 +2,429 @@
-
-    Example
-    -------
-    Consider dimension (n) = 2, and exactness (\mu) = 1 and
-    Chebyshev's polynomial of first kind, and its extremums as
-    the basis function and points, then:
-    ..math::
-        levels in one-dimension = [[0], [1,2]]
-        indices = [[1,1], [1,2], [2,1]]
-    replacing each indicy with corresponding indexes determined by levels:
-    ..math::
-        level's combination = [([0],[0]), ([0],[1,2]), ([1,2],[0])]
-        then:
-        grid points indexes = [(0,0), (0,1), (0,2), (1,0), (2,0)]
-        points in one-dimension = [0, -1, 1]
-        basis in one-dimension = [:class:`ChebyshevFirstKind`,
-                :class:`ChebyshevFirstKind`, :class:`ChebyshevFirstKind`]
-
-    Then:
-    ..math:
-        grid points = [(0, 0), (0, -1), (0, 1), (-1, 0), (1, 0)]
-        levels in two dimension: [(0, 0), (0, 1), (0, 1), (1, 0), (1, 0)]
-        grid points' functions = [(:class:`ChebyshevFunction`,
-        :class:`ChebyshevFunction`)
-        , (:class:`ChebyshevFirstKind`, :class:`ChebyshevFirstKind`),
-        (:class:`ChebyshevFirstKind`, :class:`ChebyshevFirstKind`),
-        (:class:`ChebyshevFirstKind`, :class:`ChebyshevFirstKind`),
-        (:class:`ChebyshevFirstKind`, :class:`ChebyshevFirstKind`)]
+    number_points: int
+        number of points to generate
+
+    method: {"uniform", "halton", "sobol", "latin"}
+        the method of generating random points
+
+    seed: {int, numpy.random.Generator}
+        seed for generating the random points
 
+    options: {None, dict}, optional
+        if method != None, additional parameters passed to the QMCEngine
     """
 
-    def __init__(self, basis_set):
-        super().__init__(basis_set)
-        if not isinstance(self.basis_set, NestedBasisFunctionSet):
-            raise TypeError('Basis set must be nested for Smolyak grid')
+    def __init__(self, domain, number_points, method, seed, options=None):
+        super().__init__()
+        self.domain = domain
+        self.number_points = number_points
+        self.method = method
+        self.seed = seed
+        self.options = options
 
-    def __call__(self, dimension):
-        """Make grid points and their corresponding basis functions.
+    @property
+    def valid_methods(self):
+        """list: the allowed methods for generating random numbers"""
+        return ["latin", "halton", "sobol", "uniform"]
 
-        Depending on the dimensionality, a set of grid points are generated
-        (integer grid points). Grid points of the basis function, and
-        their corresponding basis function and levels are then
-        generated by replacing the indexes with their
-        corresponding unidimensional points, functions, and levels' indexes.
+    @property
+    def domain(self):
+        """numpy.ndarray: domain of the point set."""
+        if self.dimension > 1:
+            return numpy.array(self._domain)
+        else:
+            return numpy.array(self._domain[0])
```

Comment:
If these are multidimensional point sets, I don't think we should flatten the shape of the domain. You can make unidimensional point sets if you want to have access to random point sets that behave like this.